app/src/main/java/org/schabi/newpipe/streams/io/NoFileManagerSafeGuard.java

```diff
@@ -0,0 +1,67 @@
+package org.schabi.newpipe.streams.io;
+
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.util.Log;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+
+import org.schabi.newpipe.R;
+
+/**
+ * Helper for when no file-manager/activity was found.
+ */
+public final class NoFileManagerSafeGuard {
+    private NoFileManagerSafeGuard() {
+        // No impl
+    }
+
+    /**
+     * Shows an alert dialog when no file-manager is found.
+     * @param context Context
+     */
+    private static void showActivityNotFoundAlert(@NonNull final Context context) {
+        if (context == null) {
+            throw new IllegalArgumentException(
+                    "Unable to open no file manager alert dialog: Context is null");
+        }
```

Comment:
Can you even pass in null, if you annotate it as `@NonNull`?

Reply:
https://stackoverflow.com/a/35604877

If you _send_ ``null`` it throws an exception.

Reply:
Doesn't it warn in Android Studio though?

Reply:
I don't quite like it coming from the Kotlin world, but I guess thats just how Java is 😕